### PR TITLE
Add external media uploader to media actions

### PR DESCRIPTION
**Pull request by johngodley**
Adds an external media service uploader to the file uploader.

This allows the same base code to be used, but sends the data to the external media service endpoint instead of the normal media endpoint.

## How to test

Run the included unit tests. Manually verify that media uploads and 'add url' continue to work.

### CODE REVIEW

**Comment by nosolosw** [comment]
Uploader works in both Jetpack and non-Jetpack sites (although Jetpack ones cannot upload images by URL, it is reported in https://github.com/Automattic/jetpack/issues/7339) from the modal and the media library.

Tests for `client/lib/media` pass. When I try `npm run test-client` I've got a out of memory error, but I've got the same in `master` branch, so I don't think this is related.

Will take a look at the code.

File: client/lib/media/test/utils.js [hunk 1]
```diff
@@ -1,18 +1,59 @@
 /**
  * External dependencies
  */
-var expect = require( 'chai' ).expect,
-	map = require( 'lodash/map' ),
-	useFakeDom = require( 'test/helpers/use-fake-dom' );
+import { expect } from 'chai';
```
Comment (nosolosw) [inline client/lib/media/test/utils.js]: :heart:

File: client/lib/media/utils.js [hunk 2]
```diff
@@ -535,6 +535,14 @@ const MediaUtils = {
 				extension: MediaUtils.getFileExtension( file ),
 				mime_type: MediaUtils.getMimeType( file )
 			} );
+		} else if ( file.thumbnails ) {
```
Comment (nosolosw) [inline client/lib/media/utils.js]: Would you mind explaining what's going on here? I've reviewed the code but it's not clear to me how whether the file we are trying to upload has thumbnails or not is an indication of it being uploaded from an external source. Is this something related to a specific service?

Comment (johngodley) [inline client/lib/media/utils.js]: Use of this was actually in another PR, so I've removed it from this PR and I'll include it there -
 it's a more suitable place to show the whole flow.
